db/db_impl/db_impl.cc

```diff
@@ -6040,19 +6039,24 @@ Status DBImpl::IngestExternalFiles(
         for (size_t i = 0; i != num_cfs; ++i) {
           if (need_flush[i]) {
             mutex_.Unlock();
-            auto* cfd =
-                static_cast<ColumnFamilyHandleImpl*>(args[i].column_family)
-                    ->cfd();
-            status = FlushMemTable(cfd, flush_opts,
-                                   FlushReason::kExternalFileIngestion,
-                                   true /* entered_write_thread */);
+            status =
+                FlushMemTable(ingestion_jobs[i].GetColumnFamilyData(),
+                              flush_opts, FlushReason::kExternalFileIngestion,
+                              true /* entered_write_thread */);
             mutex_.Lock();
             if (!status.ok()) {
               break;
             }
           }
         }
       }
+      if (status.ok()) {
+        for (size_t i = 0; i != num_cfs; ++i) {
+          if (immutable_db_options_.atomic_flush || need_flush[i]) {
+            ingestion_jobs[i].SetFlushedBeforeRun();
```

Comment:
Can we move each ingestion_jobs[i].SetFlushedBeforeRun();  right after knowing the flush status of each job so we don't need "if (immutable_db_options_.atomic_flush || need_flush[i])]" here?

Reply:
For atomic flush, there is only one flush call, so this is gonna add a for loop after the `AtomicFlushMemTables` call, right?

Reply:
yeah ... I thought that would be fine. But now that I look at it again, either way of writing are fine - not a big deal :)